src/DataCite.jl

```diff
@@ -0,0 +1,52 @@
+struct DataCite <: DataRepo
+end
+
+function description(repo::DataCite, mainpage)
+    attributes = mainpage["attributes"]
+    desc = attributes["description"]
+    authors = join.([[names[2] for names in value] for value in attributes["author"]], " ")
+    author = format_authors(authors)
+    license = attributes["license"]
+    date = attributes["published"]
+    paper = format_papers(authors, date, attributes["title"] * " [Data set]. " * attributes["container-title"] * ".", mainpage["id"])
+    
+    final = escape_multiline_string("""
+    Author: $(author)
+    License: $(license)
+    Date: $(date)
+
+    $(desc)
+
+    Please cite this paper:
+    $(paper)
+    if you use this in your research.
+    
+    """, "\$")
+end
+
+function get_urls(repo::DataCite, page)
+    urls = []
+    urls
+end
+
+function get_checksums(repo::DataCite, page)
+    nothing
+end
+
+function data_fullname(::DataCite, mainpage)
+    mainpage["attributes"]["title"]
+end
+
+function website(repo::DataCite, mainpage_url)
+    replace(mainpage_url, "https://api.datacite.org/works/", "https://doi.org/")
```

Comment:
I think you want to do

```
base_url(::DataCite) = "https://api.datacite.org/works/"
```

And use that both in `website` and `mainpage_url`.

Though don't we have a fallback for `mainpage_url` that basically does what your are doing below,
minus the validation that it is a DOI.
And doesn't that fallback use `base_url`?
In which case we can define `base_url` then remove the definition of `mainpage_url` entirely.

Reply:
Right, what about the validation then? Should I remove it then?

Reply:
I think so yes.
I'm not sure if it is actually useful right now

Reply:
So, I implemented it so that we can provide anything, maybe the search url, or datacite api url, or just the DOI. In any case, it should be able to work.